packages/react-reconciler/src/ReactFiberReconciler.js

```diff
@@ -209,6 +213,68 @@ function findHostInstance(component: Object): PublicInstance | null {
   return hostFiber.stateNode;
 }
 
+function findHostInstanceWithWarning(
+  component: Object,
+  methodName: string,
+): PublicInstance | null {
+  if (__DEV__) {
+    const fiber = ReactInstanceMap.get(component);
+    if (fiber === undefined) {
+      if (typeof component.render === 'function') {
+        invariant(false, 'Unable to find node on an unmounted component.');
```

Comment:
I know these are the same, but dev-only invariants feel risky.

Reply:
yea see what you're saying but refactoring this function is a pita and makes it worse.

Once we fix RN to not use the special NativeMixin/NativeComponent, we can remove this special case and always warn with the same function.